### ISSUE TITLE
False positive error involving double reference

### ISSUE THREAD

**Issue by Coder-256** [rust-lang/rust-analyzer#18061]
I'm getting what appears to be a false-positive error in my IDE, but `cargo build` succeeds.

**rust-analyzer version**: 0.3.2096-standalone

**rustc version**: rustc 1.81.0 (eeb90cda1 2024-09-04)

**editor or extension**: VS Code 1.93.0, rust-analyzer VS Code extension v0.3.2096

**relevant settings**: none

**repository link (if public, optional)**: N/A

**code snippet to reproduce**: 
```rust
use std::borrow::Borrow;

fn buggy<T: Borrow<Q>, Q: Ord>(a: T, b: &Q) -> std::cmp::Ordering {
    a.borrow().cmp(b)
}
```

Error:

```
expected &&{unknown}, found &Q
```

Screenshot:
<img width="483" alt="Screenshot 2024-09-05 at 9 06 51 PM" src="https://github.com/user-attachments/assets/92c48fb2-be16-4952-8bfa-525a83e3e372">

Ironically if I *do* add a reference, it triggers a clippy warning ("this expression creates a reference which is immediately dereferenced by the compiler").

**Comment by ChayimFriedman2** [comment]
Probably #5514.

**Comment by Veykril** [comment]
Dupe of https://github.com/rust-lang/rust-analyzer/issues/5514
